Text/JSON/Canonical/Types.hs

```diff
@@ -26,7 +26,7 @@ import Data.Ix (Ix)
 #if !MIN_VERSION_base(4,8,0)
 import Data.Monoid (Monoid)
 #endif
-#if MIN_VERSION_base(4,9,0)
+#if !MIN_VERSION_base(4,9,0)
```

Comment:
I don't understand this proposed change.

Isn't that when semigroup was introduced? And what does it have to do with changes in GHC 9.2 or 9.4?

Is it deliberate? There's no change to the corresponding use below.

Reply:
I got a redundant import so I flipped the logic and it compiled. I meant to come back to this but forgot.

What versions of GHC is this package supposed to support? If its just `8.10` this can just be dropped. If its earlier this project does no have any CI and I do not have a compiler earlier that `8.10.7`.

Reply:
If we just need to support `8.10` and later, then we can just remove the import. If we need to support earlier versions then something like this should work:
```
+#if (MIN_VERSION_base(4,9,0) && !MIN_VERSION_base(4,13,0))
 import Data.Semigroup (Semigroup)
 #endif
```
but I am not exactly sure about `MIN_VERSION_base(4,13,0)` and I have no way to test it.